src/Components/NavbarHorizontal.php

```diff
@@ -416,35 +414,81 @@ protected function getPersonalTools() {
 		if ($widgets) {
 			$ret .=
 				$this->indent() . '<!-- personal widgets -->' .
-				$widgets ;
+				$widgets	.
+				//Parametre menu
+				$this->indent() . '<ul class="navbar-tools navbar-nav" >' .
+				$this->indent( 1 ) . '<li class="dropdown navbar-tools-tools">'	.
+				$this->indent( 1 ) . '<a class="dropdown-toggle ' . $toolsClass . '" href="#" data-toggle="dropdown" title="Paramètre" > <span class="glyphicon glyphicon-cog"></span> </a>' .
+				$this->indent() . '<ul class="p-personal-tools dropdown-menu dropdown-menu-right" >'.
+				$this->indent( 1 );
+				foreach ( $this->getSkinTemplate()->getPersonalTools() as $key => $item ) {
+					if ( $item["links"][0]["class"] == "param-menu" ) {
+						$ret .= $this->indent() . $this->getSkinTemplate()->makeListItem( $key, $item );
+					}
+				}
+				// Sub menu parameter
+				$this->indent(1) .
+				$ret .= " <hr style='border-bottom-style: solid;margin-top: 2px;margin-bottom: 2px'> " ;
+				foreach ( $this->getSkinTemplate()->getPersonalTools() as $key => $item ) {
+					if ( $item["links"][0]["class"] == "param-sub-menu" ) {
+						$ret .= $this->indent() . $this->getSkinTemplate()->makeListItem( $key, $item );
+					}
+				}
+				$ret .=
+				$this->indent( -1 ) . '</li>' .
+				$this->indent( -1 ) . '</ul>'.
+				$this->indent( -1 ) . '</ul>';
+				//End of menu parameter
 		}
 
 		// start personal tools element
-
+	
 		$ret .=
 			$this->indent() . '<!-- personal tools -->' .
 			$this->indent() . '<ul class="navbar-tools navbar-nav" >' .
 			$this->indent( 1 ) . '<li class="dropdown navbar-tools-tools">' .
 			$this->indent( 1 ) . '<a class="dropdown-toggle ' . $toolsClass . '" href="#" data-toggle="dropdown" title="' . $toolsLinkText . '" >' . $linkText . '</a>' .
-			$this->indent() . '<ul class="p-personal-tools dropdown-menu dropdown-menu-right" >';
+			$this->indent() . '<ul class="p-personal-tools dropdown-menu dropdown-menu-right" >'.
+			$this->indent( 1 );
 
-		$this->indent( 1 );
-
-
-		// add personal tools (links to user page, user talk, prefs, ...)
 		foreach ( $this->getSkinTemplate()->getPersonalTools() as $key => $item ) {
-			if(in_array($key, $personnalsToolsWidgets)) {
+			// if user isn't log show menu special loggin
+			if ( $user->isLoggedIn() == false && $key !== "language" && $key !== "help-target" ) {
+				$ret .= $this->indent() . $this->getSkinTemplate()->makeListItem( $key, $item );
 				continue;
 			}
-
-			$ret .= $this->indent() . $this->getSkinTemplate()->makeListItem( $key, $item );
+			// if link as class user-menu
+			if ( $item["links"][0]["class"] == "user-menu" ) {
+				//add icone 
+				if ( $key == "saved-drafts") {
```

Comment:
tu a laissé ici un clef spécifique, alors qu'on aurai pu faire plus générique

par exemple : if ( isset($item['icon']) ) 
ca requiert de modifier l'extension draft, ou dokit-user-menu, pour mettre cette clefs, mais ca serai plus générique et permettrai d'ajouter des icon sur d'autre lien, sans ajouter de if dépendant de la clefs

Reply:
`				if ( isset( $item['links'][0]['icon'] ) ) {
					$icon = $item['links'][0]['icon'];
					$savedLink = $this->indent() . $this->getSkinTemplate()->makeListItem( $key, $item );
					$savedLink = str_replace("</a>",$icon."</a>",$savedLink);
					$ret .= $savedLink;
					continue;
				}`